.github/workflows/build.yml

```diff
@@ -5,10 +5,11 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
       - name: Install required packages
         run: |
+          sudo apt update && sudo apt upgrade
```

Comment:
I believe `-y` is required for the upgarde operation. Also, `apt` commands have an [unstable API](https://manpages.debian.org/unstable/apt/apt.8.en.html), `apt-get` should be used in scripts.